lib/util/Components.js

```diff
@@ -258,6 +258,8 @@ function componentRule(rule, context) {
       var returnsReactCreateElement =
         node[property] &&
         node[property].callee &&
+        node[property].callee.object &&
+        node[property].callee.object.name === 'React' &&
```

Comment:
won't this prevent `import { createElement } from 'react';` from being detected?